portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```
@@ -88,9 +97,37 @@ public int getMaxComments(HttpServletRequest request) {
       return -1;
     }
 
-    // Preconditions.checkArguments(maxComments == -1 || maxComments < 5 || maxComments > 10);
+    Preconditions.checkArgument(maxComments == 5 || maxComments == 10, "%s: not a valid value.");
 
     return maxComments;
   }
 
+  /** Retreives the Page parameter and converts it to int */
+  public int getPage(HttpServletRequest request) {
+    String pageString = request.getParameter("page");
+
+    int page;
+    try {
+      page = Integer.parseInt(pageString);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + pageString);
```

instead of error logging, this code should just throw a IllegalArgumentArrer here.